jujugui/static/gui/src/app/app.js

```diff
@@ -1727,7 +1727,16 @@ YUI.add('juju-gui', function(Y) {
       // route dispatch proceed. On env connection change, the app will
       // re-dispatch and this route callback will be executed again.
       if (!this.env || !this.env.get('connected')) {
-        return;
+        // If connected to a jem, we may be in a disconnected state while
+        // waiting for the new model to be defined as part of the deployment
+        // flow. In that case, carry on as normal.
+        if (this.jem) {
+          this.set('loggedIn', false);
+          this.maskVisibility(false);
+          next();
+        } else {
```

Comment:
This else block can be removed and the surrounding conditional (!this.env) should simply end in `return;` That way it maintains it's original functionality while special casing the (this.jem) case.